### DISCUSSION TITLE
Custom external db user & permissions to that table

### DISCUSSION THREAD

**Discussion by kkmehta03** [Budibase/budibase#10524] [Help] [answered]
I have a postgreSQL database where there are users and user permission for specific tables. 
I want to apply those same rules to Budibase. 
E.g. 
I have a table called Library which has a foreign key to postgreSQL's user table, where the user is the library maintainer. 
This user only has access to that library. Now I want this functionality, where the library maintainer can login to Budibase and only perform CRUD on his library, and not any other table instances.
Meaning that library maintainer shouldn't be able to access other libraries.
How can this be done?

**Comment by melohagan** [comment, marked as answer]
Hey @kkmehta03 

You can manage the access levels for your Library table, similar to what is shown here: https://docs.budibase.com/docs/ms-sql-server#adding-a-public-products-catalogue-screen

This involves [app specific roles](https://docs.budibase.com/docs/user-roles#app-specific-roles) and you can add custom roles as well.

It is important to note however that currently roles are hierarchical, which means you cannot have segregated roles per table. Each role inherits from another.
I have raised a feature request here: https://github.com/Budibase/budibase/issues/10645
